### DISCUSSION TITLE
CoordSys3D vs. ReferenceFrame

### DISCUSSION THREAD

**Discussion by deadbeefv** [sympy/sympy#24674] [Q&A] [answered]
I am working on a dynamics problem where I need to define two reference frames (Inertial and Body-fixed). Apart from rotating the body-fixed coordinate system relative to the inertial coordinate system, I also need to translate the former relative to the latter. Using the  mechanics module (sympy.physics.mechanics) I can define the transformations between the two reference frames, but cannot translate/offset the body-fixed coord system. I have come up with two solutions but cannot proceed because they are not documented.

1. Use the CoordSys3D (from the sympy.vector module) and one of the orienter classes to specify the location of the new reference frame as a vector. The question I have for this solution is if it is possible to interchangeably use this module with the mechanics module to do my modelling and what pitfalls should I watch out for?
2. Use 3D Poses (rotation + translation). Sympy does not seem to have this implemented (at my current depth in the package's documentation rabbit hole). How could I go about this without losing the flexibility afforded by the abstractions implemented by the package?

**Comment by moorepants** [comment, marked as answer]
sympy.vector and sympy.physics.vector are not interoperable in any useful way.

sympy.physics.vector ReferenceFrames are only for tracking relative orientations, there is no concept of a translating reference frame. Point objects are what manage translation. The dynamical formulation in the module isn't built on the ideas you find in graphics (i.e. transformation matrix that can manage rotation of a frame and translation of a point in one mathematical element).

If you want to use the mechanics package, it is best to follow the formalism it uses: ReferenceFrame for rotation and Point for translation.

My book explains some of these things: https://moorepants.github.io/learn-multibody-dynamics/

**Reply by moorepants** [reply]
But we do have the sympy.physics.joints module which is newer. It has the concept of a Body which has both rotation and translation. Maybe that's helpful.

**Comment by deadbeefv** [comment]
Thank you very much for the quick response. I am actually taking your class (2021/2022 version) and it is what inspired me to pursue my project.
